libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/modifiers/Keyboard.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2025 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
+ * Please see LICENSE files in the repository root for full details.
+ */
+
+package io.element.android.libraries.designsystem.modifiers
+
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.input.key.Key
+import androidx.compose.ui.input.key.KeyEventType
+import androidx.compose.ui.input.key.isShiftPressed
+import androidx.compose.ui.input.key.key
+import androidx.compose.ui.input.key.onKeyEvent
+import androidx.compose.ui.input.key.type
+
+/**
+ * Modifier to handle Shift + F10 key events.
+ * This is typically used to trigger context menus in desktop applications.
+ *
+ * @param onShiftF10Press The callback to invoke when Shift + F10 is pressed.
+ */
+fun Modifier.onShiftF10(
```

Comment:
I kind of like the name, but it's a bit odd to find 'shiftF10` as a modifier. Could we maybe rename this to `onKeyboardContextMenuAction` maybe? Not a big deal if you'd rather keep the original name.